packages/react-server/src/ReactFizzNewContext.js

```diff
@@ -244,7 +245,14 @@ export function popProvider<T>(context: ReactContext<T>): ContextSnapshot {
     }
   }
   if (isPrimaryRenderer) {
-    prevSnapshot.context._currentValue = prevSnapshot.parentValue;
+    const value = prevSnapshot.parentValue;
+    if (value === REACT_SERVER_CONTEXT_DEFAULT_VALUE_NOT_LOADED) {
```

Comment:
Even though this isn't necessary right now since we can't pass Context through context (yet), once we add that, during a refetch we'd be able to send a context back up to the server and `useContext` it before its initialized. We should add a warning though if it was never initialized at all though, same thing could happen on the client once we add the ability to send the context itself.